app/components/layouts/BackNavigation.tsx

```diff
@@ -0,0 +1,27 @@
+import React from "react";
+
+import { Button } from "components/ui/inline/Button/Button";
+import { useHistory } from "react-router-dom";
+
+// Renders a smart back link that handles pages visited directly or from referring page
+export const BackNavigation = ({
+  defaultReturnTo,
+}: {
+  defaultReturnTo: string;
+}) => {
+  const history = useHistory();
+
+  // The "POP" action is reserved as a default action for newly created history objects. `react-router-dom` depends on
+  // another history library that defines this behavior. Read more:
+  // https://github.com/remix-run/history/blob/main/docs/api-reference.md#reference
+  const backDestination =
+    history.action === "POP"
+      ? () => history.push(defaultReturnTo)
+      : () => history.goBack();
+
+  return (
+    <Button onClick={backDestination} variant="linkWhite">
```

Comment:
Looks like the design mockup shows a left arrow icon before the text, adding the prop type `arrowVariant="before"` to the Button should fix that

Reply:
Nice 👀 !